pkg/config/nodetreemodel/config.go

```diff
@@ -91,29 +97,42 @@ func (c *ntmConfig) getValue(key string) (interface{}, error) {
 	return c.leafAtPath(key).GetAny()
 }
 
-func (c *ntmConfig) setValueSource(key string, newValue interface{}, source model.Source) {
+func (c *ntmConfig) setValueSource(key string, newValue interface{}, source model.Source) { // nolint: unused // TODO fix
 	err := c.leafAtPath(key).SetWithSource(newValue, source)
 	if err != nil {
 		log.Errorf("%s", err)
 	}
 }
 
+func (c *ntmConfig) set(key string, value interface{}, tree InnerNode, source model.Source) (bool, error) {
+	parts := strings.Split(strings.ToLower(key), ",")
+	return tree.SetAt(parts, value, source)
+}
+
 // Set assigns the newValue to the given key and marks it as originating from the given source
 func (c *ntmConfig) Set(key string, newValue interface{}, source model.Source) {
-	if source == model.SourceDefault {
-		c.SetDefault(key, newValue)
-		return
+	var tree InnerNode
+
+	// TODO: have a dedicated mapping in ntmConfig instead of a switch case
```

Comment:
I think it should be considered a bug if this method is being called with a source equal to `Default` or `File`. Those sources should only be built by `SetDefault` + `SetKnown` + `BindEnv` for default, and by `ReadConfig` for file.

Reply:
Yes, I'll update the comment. For now it's useful to have this method until we have `SetDefault` working.